File: classical-components-demo/mrz-scanner/src/main/java/io/scanbot/example/MRZResultActivity.java
package io.scanbot.example;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import io.scanbot.mrzscanner.model.MRZCheckDigit;
import io.scanbot.mrzscanner.model.MRZRecognitionResult;

public class MRZResultActivity extends AppCompatActivity {

    private static final String EXTRA_MRZ_RESULT = "MRZ_RESULT";

    public static Intent newIntent(final Context context, final MRZRecognitionResult result) {
        final Intent intent = new Intent(context, MRZResultActivity.class);
        intent.putExtra(EXTRA_MRZ_RESULT, result);
        return intent;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_mrz_result);

        TextView travelDocType = findViewById(R.id.travelDocType);
        TextView documentCode = findViewById(R.id.document_code);
        TextView firstName = findViewById(R.id.first_name);
        TextView lastName = findViewById(R.id.last_name);
        TextView issuingStateOrOrganization = findViewById(R.id.issuingStateOrOrganization);
        TextView departmentOfIssuance = findViewById(R.id.departmentOfIssuance);
        TextView nationality = findViewById(R.id.nationality);
        TextView dateOfBirth = findViewById(R.id.dateOfBirth);
        TextView gender = findViewById(R.id.gender);
        TextView dateOfExpiry = findViewById(R.id.dateOfExpiry);
        TextView personalNumber = findViewById(R.id.personalNumber);
        TextView optional1 = findViewById(R.id.optional1);
        TextView optional2 = findViewById(R.id.optional2);
        TextView discreetIssuingStateOrOrganization = findViewById(R.id.discreetIssuingStateOrOrganization);
        TextView validCheckDigitsCount = findViewById(R.id.validCheckDigitsCount);
        TextView checkDigitsCount = findViewById(R.id.checkDigitsCount);
        TextView checkDigits = findViewById(R.id.checkDigits);

        final MRZRecognitionResult result = getIntent().getParcelableExtra(EXTRA_MRZ_RESULT);

        travelDocType.setText(result.travelDocType.name());
        documentCode.setText(result.documentCodeField().value);
        firstName.setText(result.firstNameField().value);
        lastName.setText(result.lastNameField().value);
        issuingStateOrOrganization.setText(result.issuingStateOrOrganizationField().value);
        departmentOfIssuance.setText(result.departmentOfIssuanceField().value);
        nationality.setText(result.nationalityField().value);
        dateOfBirth.setText(result.dateOfBirthField().value);
        gender.setText(result.genderField().value);
        dateOfExpiry.setText(result.dateOfExpiryField().value);
        personalNumber.setText(result.personalNumberField().value);
        optional1.setText(result.optional1Field().value);
        optional2.setText(result.optional2Field().value);
        discreetIssuingStateOrOrganization.setText(result.discreetIssuingStateOrOrganizationField().value);
        validCheckDigitsCount.setText("" + result.validCheckDigitsCount);
        checkDigitsCount.setText("" + result.checkDigitsCount);

        String checkDigitsOutput = "";
        for (final MRZCheckDigit cd: result.checkDigits) {
            checkDigitsOutput += cd.type.name() + ": " + (char)cd.checkDigitCharacter + " (" + cd.successfullyValidated + ")";
            checkDigitsOutput += "\n";
        }
        checkDigits.setText(checkDigitsOutput);

        findViewById(R.id.retry).setOnClickListener(v -> finish());
    }
}

File: classical-components-demo/mrz-scanner/src/main/java/io/scanbot/example/MRZResultActivity.kt
package io.scanbot.example

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import io.scanbot.mrzscanner.model.MRZRecognitionResult

class MRZResultActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_mrz_result)

        val travelDocType = findViewById<TextView>(R.id.travelDocType)
        val documentCode = findViewById<TextView>(R.id.document_code)
        val firstName = findViewById<TextView>(R.id.first_name)
        val lastName = findViewById<TextView>(R.id.last_name)
        val issuingStateOrOrganization = findViewById<TextView>(R.id.issuingStateOrOrganization)
        val departmentOfIssuance = findViewById<TextView>(R.id.departmentOfIssuance)
        val nationality = findViewById<TextView>(R.id.nationality)
        val dateOfBirth = findViewById<TextView>(R.id.dateOfBirth)
        val gender = findViewById<TextView>(R.id.gender)
        val dateOfExpiry = findViewById<TextView>(R.id.dateOfExpiry)
        val personalNumber = findViewById<TextView>(R.id.personalNumber)
        val optional1 = findViewById<TextView>(R.id.optional1)
        val optional2 = findViewById<TextView>(R.id.optional2)
        val discreetIssuingStateOrOrganization = findViewById<TextView>(R.id.discreetIssuingStateOrOrganization)
        val validCheckDigitsCount = findViewById<TextView>(R.id.validCheckDigitsCount)
        val checkDigitsCount = findViewById<TextView>(R.id.checkDigitsCount)
        val checkDigits = findViewById<TextView>(R.id.checkDigits)
        val result: MRZRecognitionResult? = intent.getParcelableExtra(EXTRA_MRZ_RESULT)

        result?.let {
            travelDocType.text = result.travelDocType.name
            documentCode.text = result.documentCodeField().value
            firstName.text = result.firstNameField().value
            lastName.text = result.lastNameField().value
            issuingStateOrOrganization.text = result.issuingStateOrOrganizationField().value
            departmentOfIssuance.text = result.departmentOfIssuanceField().value
            nationality.text = result.nationalityField().value
            dateOfBirth.text = result.dateOfBirthField().value
            gender.text = result.genderField().value
            dateOfExpiry.text = result.dateOfExpiryField().value
            personalNumber.text = result.personalNumberField().value
            optional1.text = result.optional1Field().value
            optional2.text = result.optional2Field().value
            discreetIssuingStateOrOrganization.text = result.discreetIssuingStateOrOrganizationField().value
            validCheckDigitsCount.text = result.validCheckDigitsCount.toString()
            checkDigitsCount.text = result.checkDigitsCount.toString()

            var checkDigitsOutput = ""
            for ((checkDigitCharacter, successfullyValidated, type) in result.checkDigits) {
                checkDigitsOutput += type.name + ": " + checkDigitCharacter.toChar() + " (" + successfullyValidated + ")"
                checkDigitsOutput += "\n"
            }
            checkDigits.text = checkDigitsOutput
            findViewById<View>(R.id.retry).setOnClickListener { v: View? -> finish() }
        }
    }

    companion object {
        private const val EXTRA_MRZ_RESULT = "MRZ_RESULT"
        @JvmStatic
        fun newIntent(context: Context?, result: MRZRecognitionResult?): Intent {
            val intent = Intent(context, MRZResultActivity::class.java)
            intent.putExtra(EXTRA_MRZ_RESULT, result)
            return intent
        }
    }
}